KPI/J.adoc

```diff
@@ -1,54 +1,73 @@
 == J. Broken links
 
-Detection of broken links contributing to the overall quality score.
+=== WMCP element(s)
 
-=== J.1 To which WMCP element(s) does it apply?
+Any element content with linked informations (URLs). Links are generally
+present in `<gmd:onlineResource>`, `<gmd:graphicOverview>`, 
 
-All potential linked information (URLs).
-Links are generally present in <gmd:onlineResource>, <gmd:graphicOverview>, 
-
-....
-<gmd:graphicOverview>
-        <gmd:MD_BrowseGraphic>
-          <gmd:fileName>
-            <gco:CharacterString>https://navigator.eumetsat.int/preview/0deg-snow.jpg</gco:CharacterString>
-          </gmd:fileName>
-        </gmd:MD_BrowseGraphic>
-</gmd:graphicOverview>
-....
-
-....
+```xml
 <gmd:CI_OnlineResource>
    <gmd:linkage>
       <gmd:URL>https://eumetview.eumetsat.int/mapviewer/?product=EO:EUM:DAT:MSG:SNOW</gmd:URL>
    </gmd:linkage>
 </gmd:CI_OnlineResource>
-....
+```
+
+```xml
+<gmd:graphicOverview>
+  <gmd:MD_BrowseGraphic>
+    <gmd:fileName>
+      <gco:CharacterString>https://navigator.eumetsat.int/preview/0deg-snow.jpg</gco:CharacterString>
+    </gmd:fileName>
+  </gmd:MD_BrowseGraphic>
+</gmd:graphicOverview>
+```
+
+=== What is being measured
 
-=== J.2 What is being measured?
+The number of broken links in each individual metadata record.  Broken links
+include links which, when accessed, result in a 4xx or 5xx HTTP error.footnote:[https://httpstatuses.com].
 
-The number of broken links present in each individual metadata record is being measured
+=== Rationale for measurement
 
-=== J.3 Why should we measure it?
+Broken links damage the user experience and gives the impression to users that
+a website is not maintained (88% of the online consumers are less likely to
+return to a site after a bad experience.footnote:[https://review42.com/web-design-statistics]).
+In addition, having numerous broken links affects the reputation and rank of
+your website when indexed by mass market search engines.
 
-Broken links damage the user experience and gives the impression to users that the site is not maintained (88% of the online consumers are less likely to return to a site after a bad experience). In addition broken links affects search engine rankings as having too often or too many broken links affects the reputation of your website towards Search Engines.
+=== Rules for implementation
 
-=== J.4 How is the measurement executed?
+.Broken links implementation rules
+|===
+|Rule |Score
```

Comment:
Does the number of links make it a factor for the score grading.
Your record may have one link only which is accessible , does it get a 5 because there is only one. 
Maybe the scoring can be assessed against the ratio depending on how many links the record has 1/1 (10 points)
2/4 (5 points)

Reply:
Good points @martaguti.  How about the following:

- Document has 1 or more broken links: 0
- Document has 1 - 3 accessible links: 2
- Document has 4 - 6 accessible links: 5
- Document has over 6 accessible links: 10

The idea being that even one broken link results in a 0.  Does this seem reasonable?

Reply:
I don't think this is reasonable. I don't think the score should drop to 0 if there is one bad link out of 10. Can we make this a percentage instead? 9 good out of 10 == 90%? 10 bad out of 10 == 0%?

Reply:
Good point. So do we measure number of links or percentage, or both?  Example:

- all links fail: 0
- 1 - 3 links with >= 75% pass: 2
- 4 - 6 links with >= 75% pass: 5
- 6 or more links with >= 75% pass: 10

Reply:
I don't think we should measure the count of links and the accessibility of links in one KPI. These are two different things. It's not necessarily better if there are 10 working links instead of 2 working links.

Reply:
I think I am with @amilan17  here the having more accessible links should not be a higher score factor. 
Percentage or ratio should work  IMHO 1 good out of 1 should score the same as 10 good out of 10. Otherwise people may start to put non necessary links just to get higher scores :-)

Reply:
Fair enough :)  So then what about pure percentage (i.e. number of good links as a percentage of total links)?  Else, happy to hear alternate proposals.

Reply:
FYI PR updated; I have added a relative scoring here based on number of links (total valid links / total links), for review and comment.

Reply:
Thanks, Tom. I am having troubles to navigate in the threaded discussion about this PR and the 5 (?) related commits, sorry. I am fine with the score calculated as "total valid links / total links (100%)" but I am not sure what is the final decision about HTTPS. Are HTTP links considered invalid? BTW, the check I am adding to pywcmp can correctly recognize links starting with "http://" that, on actual access, are upgraded to HTTPS.
Or are we going to give only 80% for the link validity and reserve the additional 20% as a bonus for HTTPS use? A third option would be to have link validity and HTTPS use as two separate tests.

Reply:
Thanks @josusky.  What about two separate cases foreach link?
- link resolves
- link is HTTPS

So then each link would have a maximum of two points per above, and the final score of the KPI is a percentage?  So two valid non-HTTPS links would result in 2/4, or 50%.  Two valid HTTPS links would result in 4/4, of 100%.  Does this work?  If so, I will update this KPI.

For HTTP links that, on actual access, are upgraded to HTTPS, I would not count those as HTTPS per se, as they should be represented as HTTPS from the metadata document proper.  If the HTTP links redirect to HTTP on actual access, then the metadata editor would need to update their record accordingly.